app/src/main/java/com/duckduckgo/app/buildconfig/RealAppBuildConfig.kt

```diff
@@ -65,6 +78,65 @@ class RealAppBuildConfig @Inject constructor(
     override val variantName: String?
         get() = variantManager.get().getVariantKey()
 
+    override suspend fun isAppReinstall(): Boolean = withContext(dispatcherProvider.io()) {
+        return@withContext kotlin.runCatching {
+            if (sdkInt < 30) {
+                return@withContext false
+            }
+
+            if (preferences.contains(APP_REINSTALLED_KEY)) {
+                return@withContext preferences.getBoolean(APP_REINSTALLED_KEY, false)
+            }
+
+            val isAppReinstalled = preferences.getBoolean(APP_REINSTALLED_KEY, false)
+
+            if (isAppReinstalled) {
+                return@withContext true
+            } else {
```

Comment:
**Question:** There’s already a check right before this to see if `APP_REINSTALLED_KEY` has a value, so if I’m not mistaken (though I might be, haha), we could probably remove this part?

Reply:
@nalcalag Good spot, I’ve removed that check 👍